fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/implementation/NestLoopJoinImplementationRule.java

```diff
@@ -56,6 +56,10 @@ public boolean check(final OptExpression input, OptimizerContext context) {
             if (!supportJoinType(joinType)) {
                 throw new SemanticException(UNSUPPORTED_JOIN_CLAUSE, joinType, joinOperator.getOnPredicate());
             }
+            if (!context.getSessionVariable().isEnableNestedLoopJoin()) {
+                throw new SemanticException("NestLoopJoin is not allowed. " +
+                        "Please check whether there are non-equal join conditions in the query.");
+            }
```

Comment:
It seems check both cross join and nest loop join here is more reasonable.

Reply:
@packy92 Fixed, thanks.